reposerver/repository/repository.go

```diff
@@ -1918,7 +1918,7 @@ func runConfigManagementPluginSidecars(ctx context.Context, appPath, repoPath, p
 	}
 
 	// detect config management plugin server (sidecar)
-	conn, cmpClient, err := discovery.DetectConfigManagementPlugin(ctx, appPath, pluginName, env, tarExcludedGlobs)
```

Comment:
The `DetectConfigManagementPlugin` is supposed to check the path provided in the second argument and return true if the directory contains the file that matches the settings provided in  `spec.discover` .

So if we provide `repoPath` in the second parameter then repo server would try to render manifests from e.g. `my-app` sub-directory but would try to pick a plugin based on root repo directory which is incorrect.


If I'm understanding it correctly we have performance optimization here: when repo server is trying to check if given sub-path matches plugin it sends only sub-directory content. But it causes problem with symlinks. I think only option is to send whole repo and sub-directory that should be checked. gRPC method of CMP plugin already supports it. So we just need to change [DetectConfigManagementPlugin](https://github.com/argoproj/argo-cd/blob/03513ebeec3e70926f6c33328d11be5c99478c18/util/app/discovery/discovery.go#LL88C6-L88C34) method so it would receive both `repoPath` and `appPath`

Reply:
Got it, I have updated the PR, please take a look!